### DISCUSSION TITLE
Report -> error 500

### DISCUSSION THREAD

**Discussion by registry** [Leantime/leantime#754] [Q&A] [answered]
Hi, 
i'm running Leantime on my domain
Sofar all runs good, except of reports
i get 
<domain>/reports/show  -> HTTP ERROR 500

error.log : 
[07-Jul-2022 05:15:26 America/Los_Angeles] PHP Fatal error:  Uncaught TypeError: PDOStatement::setFetchMode(): Argument #2 must be a valid class in <PATH TO Leantim>/src/domain/repo
Stack trace:
#0 <PATH TO Leantim>/src/domain/reports/repositories/class.reports.php(291): PDOStatement->setFetchMode(8, 'leantime\\domain...')
#1 <PATH TO Leantim>/src/domain/sprints/services/class.sprints.php(157): leantime\domain\repositories\reports->getSprintReport('2')
#2 <PATH TO Leantim>/src/domain/reports/controllers/class.show.php(69): leantime\domain\services\sprints->getSprintBurndown(Object(leantime\domain\models\sprints))
#3 <PATH TO Leantim>/src/core/class.frontcontroller.php(164): leantime\domain\controllers\show->get(Array)
#4 <PATH TO Leantim>/src/core/class.frontcontroller.php(107): leantime\core\frontcontroller->executeAction('reports.show')
#5 <PATH TO Leantim>/src/core/class.application.php(88): leantime\core\frontcontroller->run()
#6 <PATH TO Leantim>/public/index.php(43): leantime\core\application->start()
#7 {main}
  thrown in <PATH TO Leantim>/src/domain/reports/repositories/class.reports.php on line 291

**Comment by funkywaddle** [comment]
@registry is this still an issue? 
What release version are you using? 
Have you tried with the latest release?

**Reply by registry** [reply]
This particular prolem  resolved... i needed to add  my email in config (sick(!), not documented in ÍNSTALL instructions),
but another  problem persists, which leads to very same error.... But this tie 500 appears in 70% cases... sometimes it works... dont know the precise reason:

Error log: 

[20-Jul-2022 13:20:56 America/Los_Angeles] PHP Fatal error:  Uncaught Error: Cannot use object of type leantime\domain\models\reports as array in <path>/htdocs/management/src/domain/sprint
Stack trace:
#0 <path>/htdocs/management/src/domain/reports/controllers/class.show.php(69): leantime\domain\services\sprints->getSprintBurndown(Object(leantime\domain\models\sprints))
#1<path>/htdocs/management/src/core/class.frontcontroller.php(164): leantime\domain\controllers\show->get(Array)
#2<path>/htdocs/management/src/core/class.frontcontroller.php(107): leantime\core\frontcontroller->executeAction('reports.show')
#3 <path>/htdocs/management/src/core/class.application.php(88): leantime\core\frontcontroller->run()
#4 <path>/htdocs/management/public/index.php(43): leantime\core\application->start()
#5 {main}
  thrown in <path>/htdocs/management/src/domain/sprints/services/class.sprints.php on line 159
  
  
  i have line  /htdocs/management/src/domain/reports/controllers/class.show.php(69):
...
                          $sprintChart = $this->sprintService->getSprintBurndown($sprintObject);
...

Version: 

Powered By Leantime - 2.1.9

**Comment by marcelfolaron** [comment]
Hi,

All of these issues (including the email one) have been resolved in 2.2.0
I recommend updating to the latest version (just make sure you are running
PHP8)

**Reply by registry** [reply]
I will just need to overwrite   lean ... no need to migrate DB , and so ?

**Reply by registry** [reply]
Nope: 
updated to 2.2.0
error:
[20-Jul-2022 14:09:46 America/Los_Angeles] PHP Fatal error:  Uncaught Error: Cannot use object of type leantime\domain\models\reports as array in <path>/htdocs/management/src/domain/sprint
Stack trace:
#0 <path>/htdocs/management/src/domain/reports/controllers/class.show.php(69): leantime\domain\services\sprints->getSprintBurndown(Object(leantime\domain\models\sprints))
#1 <path>/htdocs/management/src/core/class.frontcontroller.php(191): leantime\domain\controllers\show->get(Array)
#2<path>/htdocs/management/src/core/class.frontcontroller.php(109): leantime\core\frontcontroller->executeAction('reports.show')
#3 <path>/htdocs/management/src/core/class.application.php(94): leantime\core\frontcontroller->run()
#4 <path>/htdocs/management/public/index.php(43): leantime\core\application->start()
#5 {main}
  thrown in <path>/htdocs/management/src/domain/sprints/services/class.sprints.php on line 160

**Reply by registry** [reply]
Error appears only for two out of my three projects... for one project  it wrks from some reason...

**Reply by funkywaddle** [reply]
did you try running the update? ([leantimeurl.com]/update)

**Reply by registry** [reply]
Well, i have unpacked sources over my old installation, preservind old .htaccess and configure.php  , it asked me about "DB update" -> i did,, it confirmed success , but error as posted above persists....

**Comment by marcelfolaron** [comment, marked as answer]
I’ll take a look. A new version will be released tomorrow and I’ll make
sure to include a fix.

**Reply by marcelfolaron** [reply]
Just a qiuck update. I found the issue and will release a fix today.

**Reply by registry** [reply]
with 2.2.4 seems like "fixed"

**Reply by alkisal** [reply]
Marcel hello
I am running version 2.1.9 and i have applied many updates before that without issues.
i am trying to upgrade to 2.2.4 and although i am doing everything as before i get the HTTP ERROR 500.
tried to replace .htaccess etc with the old file but nothing happens and if i remove it altogether i get the usual 403 error.
this is the first time i am facing such issues with leantime.
any suggestions please?
thanks 
Alkis

**Comment by marcelfolaron** [comment]
Hi,

Please make sure you are running PHP 8+ we dropped support for 7 as of 2.2.0

**Reply by alkisal** [reply]
Ah, I am running PHP 7.4. I will rectify and revert
thanks for your swift response as always
BR
